src/v/cloud_roles/apply_gcp_credentials.cc

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "apply_gcp_credentials.h"
+
+cloud_roles::apply_gcp_credentials::apply_gcp_credentials(
+  cloud_roles::gcp_credentials credentials)
+  : _oauth_token{fmt::format("Bearer {}", credentials.oauth_token())} {}
+
+std::error_code cloud_roles::apply_gcp_credentials::add_auth(
+  http::client::request_header& header) const {
+    auto token = _oauth_token();
+    header.insert(
+      boost::beast::http::field::authorization, {token.data(), token.size()});
+    return {};
+}
+
+void cloud_roles::apply_gcp_credentials::reset_creds(
+  cloud_roles::credentials creds) {
+    if (!std::holds_alternative<gcp_credentials>(creds)) {
+        throw std::runtime_error(fmt_with_ctx(
```

Comment:
yeah this is more or less a programmer error (calling a function with the wrong variant kind). we use `ss::visit` to avoid these issues, this guard is just an extra check.